cylc/flow/workflow_files.py

```diff
@@ -600,7 +601,13 @@ def is_installed(path: Union[Path, str]) -> bool:
     return cylc_install_folder.is_dir() and source.is_symlink()
 
 
-def _clean_check(reg: str, run_dir: Path) -> None:
+async def get_contained_workflows(path: Path) -> List[str]:
+    """Return the sorted names of any workflows in the path provided"""
+    from cylc.flow.network.scan import scan
+    return sorted([i['name'] async for i in scan(scan_dir=path)])
```

Comment:
I suspect we'd get a circular import error if this import was at the top of the file.

We kind of need to separate out the different parts of `workflow_files` into different files

Reply:
By the way, should we increase the scan depth in this case (default is 4 levels deep)? Seeing as this is a safety feature?